strapi/strapi
# Fix: GraphQL validates repeatable component attributes correctly

### What does it do?

Fixes a typo in the graphql component resolver

### Why is it needed?

Query arguments for graphql components were targeting the parent model for validation rather than the component model

### How to test it?

- create a content type with a repeatable component
- the component should have an attribute with a name not present in the content type
- run a query with a filter on that attribute of the component
- before this PR: an error would have been thrown about an invalid parameter
- with this PR: the attribute should be accepted
 
Also added a simple test that fails without the PR and passes with the PR.

### Related issue(s)/PR(s)

Fixes #17995 
Related fix #18180
